geomesa-convert/geomesa-convert-common/src/test/scala/org/locationtech/geomesa/convert2/TypeInferenceTest.scala

```diff
@@ -0,0 +1,113 @@
+  val multiPolygon = WKTUtils.read(multiPolygonString)
+
+  val geometryCollectionString = "GEOMETRYCOLLECTION(POINT(4 6),LINESTRING(4 6,7 10))"
+  val geometryCollection = WKTUtils.read(geometryCollectionString)
+
+  "TypeInference" should {
+    "infer simple types" in {
+      val types = TypeInference.infer(Seq(Seq("a", 1, 1L, 1f, 1d, true)))
+      types.map(_.typed) mustEqual Seq(STRING, INT, LONG, FLOAT, DOUBLE, BOOLEAN)
+    }
+    "infer simple types from strings" in {
+      val types = TypeInference.infer(Seq(Seq("a", "1", s"${Int.MaxValue.toLong + 1L}", "1.1", "1.00000001", "true")))
+      types.map(_.typed) mustEqual Seq(STRING, INT, LONG, FLOAT, DOUBLE, BOOLEAN)
+    }
+    "infer complex types" in {
+      val types = TypeInference.infer(Seq(Seq(new Date(), Array[Byte](0), uuid)))
+      types.map(_.typed) mustEqual Seq(DATE, BYTES, UUID)
+    }
+    "infer complex types from strings" in {
+      val types = TypeInference.infer(Seq(Seq("2018-01-01T00:00:00.000Z", uuidString)))
+      types.map(_.typed) mustEqual Seq(DATE, UUID)
+    }
+    "infer geometry types" in {
+      val types = TypeInference.infer(Seq(Seq(point, lineString, polygon, multiPoint, multiLineString,
+        multiPolygon, geometryCollection))).map(_.typed)
+      types mustEqual Seq(POINT, LINESTRING, POLYGON, MULTIPOINT, MULTILINESTRING, MULTIPOLYGON, GEOMETRY_COLLECTION)
+    }
+    "infer geometry types from strings" in {
+      val types = TypeInference.infer(Seq(Seq(pointString, lineStringString, polygonString, multiPointString,
+        multiLineStringString, multiPolygonString, geometryCollectionString))).map(_.typed)
+      types mustEqual Seq(POINT, LINESTRING, POLYGON, MULTIPOINT, MULTILINESTRING, MULTIPOLYGON, GEOMETRY_COLLECTION)
+    }
+    "merge up number types" in {
+      val types = Seq(Seq(1d), Seq(1f), Seq(1L), Seq(1))
+      foreach(types.drop(0).permutations)(t => TypeInference.infer(t).map(_.typed) mustEqual Seq(DOUBLE))
+      foreach(types.drop(1).permutations)(t => TypeInference.infer(t).map(_.typed) mustEqual Seq(FLOAT))
+      foreach(types.drop(2).permutations)(t => TypeInference.infer(t).map(_.typed) mustEqual Seq(LONG))
+    }
+    "merge up geometry types" in {
+      val types = Seq(Seq(point), Seq(lineString), Seq(polygon), Seq(multiPoint), Seq(multiLineString),
+        Seq(multiPolygon), Seq(geometryCollection))
+      foreach(types.permutations)(t => TypeInference.infer(t).map(_.typed) mustEqual Seq(GEOMETRY))
+    }
+    "merge up null values" in {
+      val values = Seq("a", 1, 1L, 1f, 1d, true, new Date(), Seq[Byte](0), uuid, point, lineString,
+        polygon, multiPoint, multiLineString, multiPolygon, geometryCollection)
+      foreach(values) { value =>
+        TypeInference.infer(Seq(Seq(value), Seq(null))) mustEqual TypeInference.infer(Seq(Seq(value)))
+      }
+    }
+    "create points from lat/lon pairs" in {
```

Comment:
Are your pairs here lat/lon or lon/lat?  If we assume X first, then the comment might need to be changed.

Reply:
fixed